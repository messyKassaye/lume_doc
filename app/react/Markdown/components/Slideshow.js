import ImageGallery from 'react-image-gallery';
import 'react-image-gallery/styles/scss/image-gallery.scss';
import PropTypes from 'prop-types';
import React from 'react';

const Slideshow = ({
  children,
  showNav,
  showFullscreenButton,
  showPlayButton,
  autoPlay,
  classname,
}) => {
  const items = children.map(c => ({ renderItem: () => c }));
  return (
    <ImageGallery
      additionalClass={classname}
      items={items}
      autoPlay={autoPlay}
      showNav={showNav}
      showThumbnails={false}
      showFullscreenButton={showFullscreenButton}
      showPlayButton={showPlayButton}
    />
  );
};

Slideshow.defaultProps = {
  autoPlay: true,
  showNav: false,
  showFullscreenButton: false,
  showPlayButton: false,
  classname: '',
};

Slideshow.propTypes = {
  classname: PropTypes.string,
  autoPlay: PropTypes.bool,
  showNav: PropTypes.bool,
  showFullscreenButton: PropTypes.bool,
  showPlayButton: PropTypes.bool,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
    PropTypes.string,
  ]).isRequired,
};

export default Slideshow;
