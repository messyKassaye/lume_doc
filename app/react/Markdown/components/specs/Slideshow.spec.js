import React from 'react';
import { shallow } from 'enzyme';
import Slideshow from '../Slideshow.js';

describe('Link', () => {
  beforeEach(() => {
    spyOn(console, 'warn');
  });
  it('should render a slideshow', () => {
    const component = shallow(
      <Slideshow>
        <img src="img1.jpg" />
        <img src="img2.jpg" />
      </Slideshow>
    );
    expect(component).toMatchSnapshot();
  });
});
