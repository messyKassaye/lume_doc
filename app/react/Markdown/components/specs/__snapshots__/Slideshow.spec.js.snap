// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should render a slideshow 1`] = `
<ImageGallery
  additionalClass=""
  autoPlay={true}
  disableArrowKeys={false}
  disableSwipe={false}
  disableThumbnailScroll={false}
  flickThreshold={0.4}
  indexSeparator=" / "
  infinite={true}
  isRTL={false}
  items={
    Array [
      Object {
        "renderItem": [Function],
      },
      Object {
        "renderItem": [Function],
      },
    ]
  }
  lazyLoad={false}
  preventDefaultTouchmoveEvent={false}
  renderFullscreenButton={[Function]}
  renderLeftNav={[Function]}
  renderPlayPauseButton={[Function]}
  renderRightNav={[Function]}
  showBullets={false}
  showFullscreenButton={false}
  showIndex={false}
  showNav={false}
  showPlayButton={false}
  showThumbnails={false}
  slideDuration={450}
  slideInterval={3000}
  startIndex={0}
  stopPropagation={false}
  swipeThreshold={30}
  swipingTransitionDuration={0}
  thumbnailPosition="bottom"
  useBrowserFullscreen={true}
  useTranslate3D={true}
/>
`;
